[NO GBP]You no longer need to change 2 other defines based on math in order to change one of the powerloss defines.

Pickle-Coding opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Removes the POWERLOSS_LINEAR_THRESHOLD and POWERLOSS_LINEAR_OFFSET defines, and instead add a constant for the supermatter that adjusts itself based on POWERLOSS_LINEAR_RATE and POWERLOSS_CUBIC_DIVISOR. Changes powerloss math to negate the need for a linear offset constant, should give the same results. Not player facing.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
It is better this way.
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

Pickle-Coding committed:
It is better this way.

Removes powerloss linear threshold and the other define. Adds a constant for powerloss that adjusts itself based on POWERLOSS_LINEAR_RATE and POWERLOSS_CUBIC_DIVISOR instead of having to adjust 2 defines by yourself. Changes powerloss function to be a sum of max and min checks instead of a ternary function, negating the need for one of the constants. Should evaluate with the same results.

LemonInTheDark commented on code/modules/power/supermatter/supermatter_process.dm:
```diff
@@ -99,7 +99,7 @@
 	if(power_changes)
 		///The power that is getting lost this tick.
-		var/power_loss = power < POWERLOSS_LINEAR_THRESHOLD ? ((power / POWERLOSS_CUBIC_DIVISOR) ** 3) : (POWERLOSS_LINEAR_OFFSET + POWERLOSS_LINEAR_RATE * (power - POWERLOSS_LINEAR_THRESHOLD))
+		var/power_loss = max(POWERLOSS_LINEAR_RATE * (power - powerloss_linear_threshold), 0) + min((power / POWERLOSS_CUBIC_DIVISOR) ** 3, (powerloss_linear_threshold / POWERLOSS_CUBIC_DIVISOR) ** 3)
```
doing it like this makes it like 2 times harder to understand, almost requires context from this change to realize that's going on.
why make it this complex?

Pickle-Coding replied:
To remove a define (or constant) that needs to get calculated.

Pickle-Coding committed:
Returns the ternary function.

Turns power_loss back into a ternary function. Adds powerloss_linear_offset constant, which gets calculated to offset the linear function to allow the transition between the cubic and linear powerloss functions to be differentiable.

LemonInTheDark commented on code/__DEFINES/supermatter.dm:
```diff
@@ -124,10 +124,6 @@
-#define POWERLOSS_LINEAR_THRESHOLD 5880.76
-/// The offset for the linear power loss function. It is the power loss when power is at POWERLOSS_LINEAR_THRESHOLD.
-#define POWERLOSS_LINEAR_OFFSET 1627.01
 /// The rate at which the linear power loss function scales with power.
```
out of interest, why turn these into variables, couldn't you embed the math operations inside them, so you don't need to change them, but they're still treated as constants?
